### PR TITLE
fix(integrations): Modifying Slack integration to prefer event data over group data

### PR DESCRIPTION
**Pull request by iProgramStuff**
Instead of always using group title, we now try to use the event title and use the group as a fallback if event is None. An addition to build_group_attachment also tries to get the latest group event if it gets called with event=None.   

Additionally, references to culprit have been removed.

Fixes SEN-882 & ISSUE-529

### CODE REVIEW

**Review by markstory — COMMENTED** [review]
Are there tests we can update to capture the new behavior, or existing tests that need to change?

**Comment by EvanPurkhiser** [comment]
I think this part of slack wasn't super deeply tested unfortunately @markstory

**Review by evanpurkhiser — APPROVED** [review]
Looks reasonable to me

File: src/sentry/integrations/slack/utils.py [hunk 1]
```diff
@@ -159,6 +153,8 @@ def build_group_attachment(group, event=None, tags=None, identity=None, actions=
 
     members = get_member_assignees(group)
     teams = get_team_assignees(group)
+    if event is None:
+        event = group.get_latest_event()
```
Comment (wedamija) [inline src/sentry/integrations/slack/utils.py]: Nit: I don't know that we need to do this? The group title should just reflect the latest event. Not blocking on this

**Review by wedamija — CHANGES_REQUESTED** [review]
I'd like to see tests around `build_attachment_title` that verify the changes. Like a group and event with different titles, and verify that the expected behaviour occurs. One without a group, etc.

File: src/sentry/integrations/slack/utils.py [hunk 2]
```diff
@@ -79,22 +79,16 @@ def get_assignee(group):
 def build_attachment_title(group, event=None):
     # XXX(mitsuhiko): This is all super event specific and ideally could just use a
     # combination of `group.title` and `group.title + group.culprit`.
```
[the remark this answers is not in the document]
Comment (iProgramStuff) [reply src/sentry/integrations/slack/utils.py]: I removed the comment. All the existing slack tests pass - going to try adding some new tests.

File: src/sentry/integrations/slack/utils.py (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/integrations/slack/utils.py]: Removed that line.

File: src/sentry/integrations/slack/utils.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (iProgramStuff) [reply src/sentry/integrations/slack/utils.py]: Added in two tests. A call to build_group_attachment with a group to get back group info, and with a group and event to get back event info.

File: src/sentry/integrations/slack/utils.py (hunk 1 above)
Comment (iProgramStuff) [reply src/sentry/integrations/slack/utils.py]: Added in two tests. A call to build_group_attachment with a group to get back group info, and with a group and event to get back event info.

File: tests/sentry/integrations/slack/test_utils.py [hunk 3]
```diff
@@ -40,3 +44,83 @@ def test_simple(self):
...
+                },
+            ],
+            "mrkdwn_in": ["text"],
+            "title": event.title,
+            "fields": [],
+            "footer": u"BENGAL-ELEPHANT-GIRAFFE-TREE-HOUSE-1",
+            "ts": to_timestamp(ts),
+            "title_link": u"http://testserver/organizations/rowdy-tiger/issues/23/?referrer=slack",
+            "callback_id": '{"issue":23}',
+            "fallback": u"[{}] {}".format(self.project.slug, event.title),
```
Comment (wedamija) [inline tests/sentry/integrations/slack/test_utils.py]: Are the group titles and event titles different here?

**Review by wedamija — APPROVED** [review]

**Pull request merged**